Angular/GridApp/src/app/relation.service.ts

```diff
@@ -0,0 +1,26 @@
+import {Injectable} from '@angular/core';
+import {HttpClient, HttpHeaders} from '@angular/common/http';
+import {Relation} from './relation';
+    
+@Injectable()
+export class RelationService{
+    
+    private url = "https://localhost:44358/api/relations";
+    constructor(private http: HttpClient){ }
+       
+    getRelations(){
+        return this.http.get(this.url);
+    }
+   
+    createRelation(relation: Relation){
+        const myHeaders = new HttpHeaders().set("Content-Type", "application/json");
```

Comment:
Change 'myHeaders' to 'headers' here and in another places. Try not to use 'helloWorldVariables'. Code like it is in a real production project.